File: scripts/opencv_01.py

import cv2 #导入OpenCv库

gray_img = cv2.imread("/home/xiehan/picture/work.png",0)
#直接在读取的时候变成单通道灰度图

save_path = "task2_final.png"#保存灰度图
cv2.imwrite(save_path,gray_img)#cv2.imwrite ：把处理好的图片保存成文件存入电脑

print("task2灰度图已保存！")#提示

cv2.namedWindow("task2")#建造窗口名称task2，有需求可以提前设置窗口大小
cv2.imshow("task2",gray_img)#展示
cv2.waitKey(0)#停留，防止一霎而过

cv2.destroyAllWindows()#关闭

File: src/opencv_01.cpp
#include <opencv2/opencv.hpp>//引入opencv核心头文件
#include <iostream>

using namespace cv;//使用cv命名空间
using namespace std;

int main() //main函数入口
{
    // 1. 读取图片（替换为你的图片路径）
    Mat img = imread("/home/xiehan/picture/work.png",0);//读取图片变成单通道灰度图，返回一个mat类型（图片容器）


    // 2. 保存+显示
    imwrite("task2_cpp.png", img);//保存图片 
    imshow("TASK2 灰度图", img);//展示图片
    waitKey(0);
    destroyAllWindows();

    return 0;
}
